src/js/components/HealthBar.js

```diff
@@ -0,0 +1,68 @@
+import classNames from 'classnames';
+import React from 'react';
+import {Tooltip} from 'reactjs-components';
+
+import HealthBarMap from '../constants/HealthBar';
+import StatusBar from './StatusBar';
+
+class HealthBar extends React.Component {
+  renderToolTip() {
+    let {tasks, instancesCount} = this.props;
+
+    tasks = Object.keys(tasks).filter(function (task) {
+      return tasks[task] !== 0 && task !== 'tasksRunning';
+    }).map(function (task, index) {
+      let percentage = parseInt(tasks[task] / instancesCount * 100, 10);
+
+      let classSet = classNames(HealthBarMap[task].cssName, 'dot icon');
+
+      return (
+        <div key={index}>
+          <span className={classSet}></span>
+          {` ${tasks[task]} ${HealthBarMap[task].label} (${percentage} %)`}
+        </div>
+      );
+    }).filter(function (item) {
+      return item != null;
+    });
+
+    if (tasks.length === 0) {
+      return 'No Running Tasks';
+    }
+
+    return tasks;
+  }
+
+  render() {
+    let {tasks, instancesCount} = this.props;
+
+    if (tasks == null) {
+      return null;
+    }
+
+    tasks = Object.keys(tasks).filter(function (task) {
+      return task !== 'tasksRunning';
+    }).map(function (taskStatus) {
+      return {className: HealthBarMap[taskStatus].cssName, value: tasks[taskStatus]};
+    });
+
+    return (
+      <Tooltip content={this.renderToolTip()}>
+        <StatusBar
+          data={tasks}
+          scale={instancesCount}/>
+      </Tooltip>
+    );
+  }
+}
+
+HealthBar.defaultProps = {
+    instancesCount: null
+};
+
+HealthBar.propTypes = {
+  instancesCount: React.PropTypes.number,
```

Comment:
Actually it is not possible. If the cluster does not have enough offers you might have a couple of instances running or staged but the sum of both might not equal the defined instances.